Assets/VRTK/Source/Scripts/Interactions/Interactables/SnapAttachMechanics/VRTK_BaseSnapAttach.cs

```diff
@@ -0,0 +1,592 @@
+                //Wipe the deprecated value after moving it over so that this code only runs once, even if scaleType is later changed from MatchScale
+                pairedSnapDropZone.applyScalingOnSnap = false;
+            }
+
+            if (pairedSnapDropZone.validObjectListPolicy != null && validObjectListPolicy == null)
+            {
+                validObjectListPolicy = pairedSnapDropZone.validObjectListPolicy;
+            }
+
+            if (defaultSnappedInteractableObject == null)
+            {
+                //When handling deprecation for the already obselete defaultSnappedObject property
+                if (pairedSnapDropZone.defaultSnappedObject != null)
+                {
+                    defaultSnappedInteractableObject = pairedSnapDropZone.defaultSnappedObject.GetComponentInParent<VRTK_InteractableObject>();
+                    if (defaultSnappedInteractableObject != null)
+                    {
+                        pairedSnapDropZone.defaultSnappedObject = null;
+                    }
+                }
+                //When handling deprecation for the defaultSnappedInteractableObject property for those users who have been on the 3.3.0 alpha branch
+                if (pairedSnapDropZone.defaultSnappedInteractableObject != null)
+                {
+                    defaultSnappedInteractableObject = pairedSnapDropZone.defaultSnappedInteractableObject.GetComponentInParent<VRTK_InteractableObject>();
+                    if (defaultSnappedInteractableObject != null) 
+                    {
+                        pairedSnapDropZone.defaultSnappedInteractableObject = null;    
+                    }
+                } 
+            }
+#pragma warning restore 618
+
+            if (!VRTK_SharedMethods.IsEditTime() && Application.isPlaying && defaultSnappedInteractableObject != null)
+            {
+                ForceSnap(defaultSnappedInteractableObject);
+            }
+        }
+
+        protected virtual void OnDisable()
+        {
+            //Stop any active animation or snapping coroutines
+            if (transitionInPlaceRoutine != null)
+            {
+                StopCoroutine(transitionInPlaceRoutine);
+            }
+
+            if (attemptTransitionAtEndOfFrameRoutine != null)
+            {
+                StopCoroutine(attemptTransitionAtEndOfFrameRoutine);
+            }
+
+            if (checkCanSnapRoutine != null)
+            {
+                StopCoroutine(checkCanSnapRoutine);
+            }
+
+            ForceUnsnap();
+            UnregisterAllGrabEvents();
+        }
+
+        /// <summary>
+        /// The IsSnapped method returns true if an Interactable Object is currently snapped to the Snap Drop Zone paired with this attach script
+        /// </summary>
+        /// <returns>Whetehr an Interactable Object is currently snapped to the Snap Drop Zone paired with this attach script.</returns>
+        public virtual bool IsSnapped()
```

Comment:
public methods should be listed before protected.

goes public, protected, private